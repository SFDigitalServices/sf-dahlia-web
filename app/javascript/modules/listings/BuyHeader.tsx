import { Heading, PageHeader, t } from "@bloom-housing/ui-components"
import React from "react"
import "./BuyHeader.scss"
import { BeforeApplyingForSale, BeforeApplyingType } from "../../components/BeforeApplyingForSale"
import { renderInlineMarkup } from "../../util/languageUtil"
import { useFeatureFlag } from "../../hooks/useFeatureFlag"

const DalpHeader = () => {
  return (
    <div className="md:bg-white md:p-4">
      <Heading styleType="underlineWeighted" className="mb-5" priority={2}>
        {t("saleDirectory.dalp.title")}
      </Heading>
      <p className="mb-4">{t("saleDirectory.dalp.content")}</p>
      <p className="font-bold">{t("saleDirectory.dalp.subtitle")}</p>
      <p className="mb-4">{t("saleDirectory.dalp.subcontent")}</p>
      <p className="mb-4">
        {renderInlineMarkup(
          t("saleDirectory.dalp.link", {
            url: "https://www.sf.gov/apply-downpayment-loan-buy-market-rate-home",
          })
        )}
      </p>
    </div>
  )
}

const BuyHeader = () => {
  const { unleashFlag: dalpDirectoryEnabled } = useFeatureFlag("temp.webapp.directory.dalp", false)
  const { unleashFlag: translationsReady } = useFeatureFlag(
    "temp.webapp.listings.sales.seeHomesForSale",
    false
  )
  return (
    <PageHeader className="buy-header">
      <div className="buy-header_columns">
        <Heading className="buy-header_title buy-header_left_col">
          {t("saleDirectory.title")}
        </Heading>
        <div className="mb-8 buy-header_right_col">
          {/* TODO: Switch this back to LinkButton or button once this issue is resolvedhttps://github.com/bloom-housing/bloom/issues/2324 */}
          <a href="#nav-bar-container" className="button is-primary is-fullwidth">
            {translationsReady
              ? t("saleDirectory.seeHomesForSale")
              : t("saleDirectory.seeTheListings")}
          </a>
        </div>
        <div className="buy-header_left_col">
          <BeforeApplyingForSale beforeApplyingType={BeforeApplyingType.DIRECTORY} />
        </div>
        {dalpDirectoryEnabled && (
          <div className="buy-header_right_col">
            <DalpHeader />
          </div>
        )}
      </div>
    </PageHeader>
  )
}

export default BuyHeader
