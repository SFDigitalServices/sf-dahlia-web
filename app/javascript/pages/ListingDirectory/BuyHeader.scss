.buy-header {
  @apply py-6;
  @apply px-4;
  @apply md:py-16;
  @apply md:px-8;
}

.buy-header_title {
  @apply mb-8;
  @apply text-4xl;
  @apply md:text-5xl;
}

.buy-header_columns {
  @apply md:grid;
  @apply md:grid-cols-12;
}
.buy-header_left_col,
.buy-header_right_col {
  @apply text-tiny;
  @apply md:text-base;
  @apply px-4;
}
.buy-header_left_col {
  @apply md:col-span-7;
}
.buy-header_right_col {
  @apply md:col-span-5;
  @apply md:col-start-8;
}
