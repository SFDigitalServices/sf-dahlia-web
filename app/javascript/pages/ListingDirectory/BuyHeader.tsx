import { Heading, PageHeader, t } from "@bloom-housing/ui-components"
import React from "react"
import Markdown from "markdown-to-jsx"
import Link from "../../navigation/Link"

const BeforeApplying = () => (
  <div className="mb-8">
    <Heading className="mb-5" priority={2} underline>
      {t("saleDirectory.beforeApplying.title")}
    </Heading>
    <p>{t("saleDirectory.beforeApplying.makeSureYou")}</p>
    <ol className="numbered-list my-6">
      <li>{t("saleDirectory.beforeApplying.step1")}</li>
      <li>
        <Markdown>
          {t("saleDirectory.beforeApplying.step2", {
            url: "https://sfmohcd.org/homebuyer-program-eligibility",
          })}
        </Markdown>
      </li>
      <li>
        <Markdown>
          {t("saleDirectory.beforeApplying.step3", {
            url: "https://sfmohcd.org/homebuyer-application-requirements#education",
          })}
        </Markdown>
      </li>
      <li>
        <Markdown>
          {t("saleDirectory.beforeApplying.step4", { url: "https://sfmohcd.org/lender-list" })}
        </Markdown>
      </li>
      <li>{t("saleDirectory.beforeApplying.step5")}</li>
    </ol>
    <Markdown>
      {t("saleDirectory.beforeApplying.readFullList", {
        url: "https://sfmohcd.org/homebuyer-program-eligibility",
      })}
    </Markdown>
  </div>
)

const GetHelp = () => (
  <div className="md:bg-white md:p-4">
    <Heading className="mb-5" priority={2} underline>
      {t("saleDirectory.getHelp.title")}
    </Heading>
    <p className="mb-4">{t("saleDirectory.getHelp.somePeopleMayQualify")}</p>
    <p className="mb-4">{t("saleDirectory.getHelp.otherGroupsMayQualify")}</p>
    <p>
      <Link external={true} href="https://sfmohcd.org/loan-programs">
        {t("saleDirectory.getHelp.viewFullListOfPrograms")}
      </Link>
    </p>
  </div>
)
const BuyHeader = () => (
  <PageHeader className="py-6 px-4 md:py-16 md:px-8">
    <div className="md:grid md:grid-cols-12">
      <Heading className="mb-8 md:col-span-7 text-4xl md:text-5xl">
        {t("saleDirectory.title")}
      </Heading>
      <div className="mb-8 md:col-span-5 md:col-start-8">
        {/* TODO: Switch this back to LinkButton or button once this issue is resolvedhttps://github.com/bloom-housing/bloom/issues/2324 */}
        <a href="#listing-results" className="button is-primary is-fullwidth">
          {t("saleDirectory.seeTheListings")}
        </a>
      </div>
      <div className="md:col-span-7">
        <BeforeApplying />
      </div>
      <div className="md:col-span-5 md:col-start-8">
        <GetHelp />
      </div>
    </div>
  </PageHeader>
)

export default BuyHeader
