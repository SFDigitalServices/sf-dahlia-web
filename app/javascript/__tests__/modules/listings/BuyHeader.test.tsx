import React from "react"
import renderer from "react-test-renderer"
import BuyHeader from "../../../modules/listings/BuyHeader"

describe("BuyHeader", () => {
  it("renders BuyHeader component", () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })
    const tree = renderer.create(<BuyHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
