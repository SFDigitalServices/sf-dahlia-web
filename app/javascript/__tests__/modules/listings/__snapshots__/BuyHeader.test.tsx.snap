// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyHeader renders BuyHeader component 1`] = `
<header
  className="page-header buy-header"
>
  <hgroup
    className="page-header__group"
  >
    <div
      className="buy-header_columns"
    >
      <h1
        className="buy-header_title buy-header_left_col"
      >
        Buy affordable housing
      </h1>
      <div
        className="mb-8 buy-header_right_col"
      >
        <a
          className="button is-primary is-fullwidth"
          href="#listing-results"
        >
          See the listings
        </a>
      </div>
      <div
        className="buy-header_left_col"
      >
        <li
          className="mb-8 style-none"
        >
          <h2
            className="text__underline-weighted mb-5 text-gray-750"
          >
            Before applying
          </h2>
          <p
            className="text-gray-750"
          >
            Make sure you:
          </p>
          <ol
            className="numbered-list my-5 text-gray-700"
          >
            <li
              className="text-gray-700"
            >
              Haven't owned residential property in the past 3 years
            </li>
            <li
              className="text-gray-700"
            >
              <p>
                Meet our 
                <a
                  href="https://sfmohcd.org/homebuyer-program-eligibility"
                  target="_blank"
                >
                  income requirements
                </a>
              </p>
            </li>
            <li
              className="text-gray-700"
            >
              <p>
                Complete 
                <a
                  href="https://sfmohcd.org/homebuyer-application-requirements#education"
                  target="_blank"
                >
                  homebuyer education
                </a>
              </p>
            </li>
            <li
              className="text-gray-700"
            >
              <p>
                Get pre-approved for a mortgage loan by a 
                <a
                  href="https://sfmohcd.org/lender-list"
                  target="_blank"
                >
                  lender on our list
                </a>
              </p>
            </li>
            <li
              className="text-gray-700"
            >
              Have enough in savings for closing costs and downpayment
            </li>
          </ol>
          <p>
            Read 
            <a
              href="https://sfmohcd.org/homebuyer-program-eligibility"
              target="_blank"
            >
              our full list of requirements
            </a>
             for more details.
          </p>
        </li>
      </div>
      <div
        className="buy-header_right_col"
      >
        <div
          className="md:bg-white md:p-4"
        >
          <h2
            className="text__underline-weighted mb-5"
          >
            Get help
          </h2>
          <p
            className="mb-4"
          >
            Some people may qualify for downpayment assistance.
          </p>
          <p
            className="mb-4"
          >
            Others, like San Francisco teachers and first responders, may qualify for loans.
          </p>
          <p>
            <a
              href="https://sfmohcd.org/loan-programs"
            >
              View the full list of programs
            </a>
          </p>
        </div>
      </div>
    </div>
  </hgroup>
</header>
`;
